src/audio/tdfb/tdfb.c

```diff
@@ -79,6 +79,187 @@ static inline int set_func(struct processing_module *mod, enum sof_ipc_frame fmt
 	return 0;
 }
 
+/*
+ * Pass-through processing functions
+ */
+
+static void tdfb_pass_same_format(struct tdfb_comp_data *cd, struct input_stream_buffer *bsource,
+				  struct output_stream_buffer *bsink, int frames)
+{
+	struct audio_stream *source = bsource->data;
+	struct audio_stream *sink = bsink->data;
+
+	audio_stream_copy(source, 0, sink, 0, frames * audio_stream_get_channels(source));
+}
+
+#if CONFIG_FORMAT_S16LE
+static void tdfb_pass_s16(struct tdfb_comp_data *cd, struct input_stream_buffer *bsource,
+			  struct output_stream_buffer *bsink, int frames)
+{
+	struct audio_stream *source = bsource->data;
+	struct audio_stream *sink = bsink->data;
+	int16_t *x = audio_stream_get_rptr(source);
+	int16_t *y = audio_stream_get_wptr(sink);
+	int16_t s0, s1;
+	int fmax;
+	int is, om;
+	int f, i, j, k;
+	const int in_nch = audio_stream_get_channels(source);
+	const int out_nch = audio_stream_get_channels(sink);
+	const int num_filters = cd->config->num_filters;
+	int remaining_frames = frames;
+
+	while (remaining_frames) {
+		fmax = audio_stream_frames_without_wrap(source, x);
+		f = MIN(remaining_frames, fmax);
+		fmax = audio_stream_frames_without_wrap(sink, y);
+		f = MIN(f, fmax);
```

Comment:
Hmm, am I missing something or can this get stuck in infinite loop if source (or sink) has less samples. Reminain_frames will never go to zero if sink/source does not have data, so this will spin indefinetely. Or is this handled somewhere?